### PR TITLE
WIP: Dropdown menu for language

### PR DESCRIPTION
**Pull request by Yashs911**
### Technical

Added a dropdown menu for the Language feature.

### Screenshot

https://user-images.githubusercontent.com/64412143/111581407-32858500-87df-11eb-848c-b301770719d1.mp4

### CODE REVIEW

**Comment by mekarpeles** [comment]
The linter is showing:

```
static/css/components/footer.less
 140:39  ✖  Unexpected leading zero                                                                         number-leading-zero      
 144:1   ✖  Expected selector ".dropdown-content a" to come before selector "#footer-icons a"               no-descending-specificity
 144:1   ✖  Expected selector ".dropdown-content a" to come before selector "#footer-links ul li a"         no-descending-specificity
 151:1   ✖  Expected selector ".dropdown-content a:hover" to come before selector "#footer-icons a"         no-descending-specificity
 151:1   ✖  Expected selector ".dropdown-content a:hover" to come before selector "#footer-links ul li a"   no-descending-specificity
 ```

**Comment by mekarpeles** [comment]
I think we probably want the style / experience to be pretty similar to http://scholar.archive.org/
scholar archive org_
scholar archive org_ (2)

**Review by jdlrobson — CHANGES_REQUESTED** [review]
Please reuse the CSS classes for the dropdowns in the menu (and markup including checkbox). Building these in the same way will cut down on css and js we use on the site and the component functionality is identical which would be a performance win!

**Review by jdlrobson — COMMENTED** [review]
Another thing to consider.. since this is a JavaScript only feature,  should we build it in Javascript rather than inside the template?

File: static/css/components/footer.less [hunk 1]
```diff
@@ -123,3 +123,45 @@ div#version-details {
   border-radius: 3px;
   padding: 8px 10px;
 }
+
+.dropbtn {
```
Comment (jdlrobson) [inline static/css/components/footer.less:127]: This feature doesn't work without javascript from what I can see and footer.less is render  blocking. Styles should be loaded inside all.js.

Comment (Yashs911) [reply static/css/components/footer.less:127]: Okay, but from what I have noticed that Header (Browse, More, and Account component) does not support keyboard navigation so is there a way to implement it in the current structure?

Comment (Yashs911) [reply static/css/components/footer.less:127]: If not I guess we should make this dropdown keeping keyboard navigation in mind.
Later we can implement the same for other dropdown menus in the header.

**Comment by Yashs911** [comment]
@jdlrobson can you please guide me on how to load styles inside all.js?

**Comment by Yashs911** [comment]
> I think we probably want the style / experience to be pretty similar to http://scholar.archive.org in the top header

@mekarpeles If I am not wrong do you want this feature in the top header of openlibrary.org or only the styles should be similar?

**Comment by jdlrobson** [comment]
Put the import inside static/css/js-all.less

**Comment by Yashs911** [comment]
@jdlrobson I made a separate less file and imported it inside js-all.less but still it's not working

**Comment by jdlrobson** [comment]
Have you run `npm run watch` ?

**Comment by Yashs911** [comment]
> Have you run `npm run watch` ?

No, only make js and make css

**Comment by jdlrobson** [comment]
`make js` is what you need to do but, bunning `npm run watch` will allow you to debug better as you don't have to run `make js` each time. For CSS loaded in JavaScript `make js` will do nothing.

I suggest adding some console.log statements in openlibrary/plugins/openlibrary/js/index.js  and checking the line `import '../../../../static/css/js-all.less';` is working by adding something obvious like `body {background: rebeccapurple !important;}` to js-all.less and the new file you are importing to debug what's going wrong for you.

If you can't work it out, feel free to push up what you have and we can work out why it's not working.

**Comment by Yashs911** [comment]
@jdlrobson Now I am confused 😕. I went to the master branch and did `make css`, `make js`, and `npm run watch`. Restarted the web container and when I click the Language option nothing changed(visually). So I checked the cookies and could see the HTTP_LANG
image
Even on the work branch (design-footer-lang), I am able to see the cookie. So I guess nothing is wrong with this PR. 
Could you please check if you can see any visual changes on the localhost without style?

File: static/css/components/footer-locale.less [hunk 2]
```diff
@@ -0,0 +1,55 @@
+.language-component__checkbox {
```
Comment (jdlrobson) [inline static/css/components/footer-locale.less:1]: Rename to language.less - footer-locale is an instance of this component and reusable

Comment (Yashs911) [reply static/css/components/footer-locale.less]: Done.

Comment (Yashs911) [reply static/css/components/footer-locale.less]: @jdlrobson strange how does renaming changes the bundlesize.
` FAIL  static/build/all.js: 128KB > maxSize 128KB (gzip) `

Comment (jdlrobson) [reply static/css/components/footer-locale.less]: #footer-locale is shorter than .language-component__checkbox and gzips better. Bundlesize is checking the size of the CSS file generated from this source code.

Comment (Yashs911) [reply static/css/components/footer-locale.less]: We have id="footer-locale-menu" but so what changes do you want me to make.

Comment (jdlrobson) [reply static/css/components/footer-locale.less]: The code here is fine. I was responding to your comment about it being stranged that it impacted bundlesize and explaining why that happens.

Comment (Yashs911) [reply static/css/components/footer-locale.less]: How should we fix this. Increase bundlesize?

Comment (Yashs911) [reply static/css/components/footer-locale.less]: @jdlrobson, if we go ahead with JS dropdown with hover fallback, then we also need to update this accordingly. So should I update it now or wait for further discussion?

**Comment by Yashs911** [comment]
https://github.com/internetarchive/openlibrary/pull/5087

**Pull request closed without merging**